### DISCUSSION TITLE
Class not found exception for LinuxFile

### DISCUSSION THREAD

**Discussion by msawicki-poland** [Pi4J/pi4j#160] [Q&A]
I created new project in IDE Netbeans (the newest) using Ant project builder because only Ant supports now remote lunching jar on raspi. Gradle and Maven in Project/Properites/Run don't have option for Runtime Platform. I downloaded libraries and added to project:
![obraz](https://user-images.githubusercontent.com/93886998/140663674-ea4cf0e5-e08b-48ad-9d33-3864796130fe.png)
and try to run (compiled ok) code:
![obraz](https://user-images.githubusercontent.com/93886998/140663707-f508b462-2e67-4088-9fb4-13a3d57e9b47.png)
but on runtime I get:
![obraz](https://user-images.githubusercontent.com/93886998/140663719-79d66b5e-57e2-497c-99dc-dada586ab036.png)

**Comment by dorjazjargal** [comment]
hi, I have the same problem as you. Did you solve this problem?If you decide, help me. /I use netbeans 12.0, java ant/

**Comment by FDelporte** [comment]
It looks the LinuxFile plugin is missing. I'm not familiar with Netbeans myself, but will try to help.

Sidenote: it is possible to create a project with IntelliJ on PC/MAC and deploy to Pi with Maven, it's both described with the Maven archetype (https://pi4j.com/getting-started/developing-on-a-remote-pc/) and CrowPi examples (https://pi4j.com/getting-started/crowpi/develop-with-intellij/).

Do you have a view of the libraries which get included in your project or uploaded to the Pi?

Instead of using `newAutoContext` can you try this approach?

```java
var piGpio = PiGpio.newNativeInstance();
var pi4j = Pi4J.newContextBuilder()
    .noAutoDetect()
    .add(
        PiGpioDigitalInputProvider.newInstance(piGpio),
        PiGpioDigitalOutputProvider.newInstance(piGpio),
        PiGpioPwmProvider.newInstance(piGpio),
        PiGpioI2CProvider.newInstance(piGpio),
        PiGpioSerialProvider.newInstance(piGpio),
        PiGpioSpiProvider.newInstance(piGpio)
    )
    .build();
```

**Comment by razorne** [comment]
Hello,

i am using I2C with pi4j for my project and run in this very same issue. 

i solved this by checking the pi4j-plugin-linuxfs.jar loaded on raspberry, and in fact LinuxFile was missing. You can open a jar just like a compressed archive, i personally use 7zip to do that. 

My guess is that during the transfer via SSH from your local machine to the raspberry something went wrong and the jar was not correctly copied, and LinuxFile was lost along the way.

As i manually replaced the jar on the raspberry, i was able to run my ant project with remote platform in netbeans and everything worked smoothly.
